Octokit/Clients/Enterprise/EnterpriseManagementConsoleClient.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Octokit
+{
+    /// <summary>
+    /// A client for GitHub's Enterprise Management Console API
+    /// </summary>
+    /// <remarks>
+    /// See the <a href="https://developer.github.com/v3/enterprise/management_console/">Enterprise Management Console API documentation</a> for more information.
+    ///</remarks>
+    public class EnterpriseManagementConsoleClient : ApiClient, IEnterpriseManagementConsoleClient
+    {
+        public EnterpriseManagementConsoleClient(IApiConnection apiConnection)
+            : base(apiConnection)
+        { }
+
+        public Uri CorrectEndpointForManagementConsole(Uri endpoint)
+        {
+            Ensure.ArgumentNotNull(endpoint, "endpoint");
+
+            if (ApiConnection.Connection.BaseAddress != null &&
+                ApiConnection.Connection.BaseAddress.ToString().EndsWith("/api/v3/", StringComparison.OrdinalIgnoreCase))
+            {
+                // We need to get rid of the /api/v3/ for ManagementConsole requests
+                // if we specify the endpoint starting with a leading slash, that will achieve this
+                return string.Concat("/", endpoint.ToString()).FormatUri();
```

Comment:
There was some issue with one of the code analysis rules not letting you form Uri's from strings but Ill take another look